build-aux/snap/snapcraft.yaml

```diff
@@ -29,173 +45,723 @@ license: GPL-3.0
 # See the comments from jdstrand in
 # https://forum.snapcraft.io/t/5547/10
 parts:
-  snapd-deb:
+  build-base:
+    plugin: nil
+    source: keyrings
+    build-packages:
+      - to i386:
+        - debootstrap
+        - fakeroot
+      - else:
+        - wget
+      - gawk
+    build-environment:
+      - RELEASE: "22.04.2"
+      - BASE: ubuntu-base-${RELEASE}-base-${CRAFT_TARGET_ARCH}.tar.gz
+      - DIR_URL: https://cdimage.ubuntu.com/ubuntu-base/releases/${RELEASE}/release
+      - URL: ${DIR_URL}/${BASE}
+      - SHA256: ${DIR_URL}/SHA256SUMS
+      - SIG: ${SHA256}.gpg
+      - DEPENDENCIES: >-
+          make
+          gcc
+          binutils
+          gawk
+          python3
+          python3-jinja2
+          meson
+          bison
+          flex
+          ninja-build
+          libc6-dev
+          gperf
+          autoconf
+          automake
+          gettext
+          libtool
+          pkg-config
+          libglib2.0-dev
+          g++
+          autopoint
+          git
+    override-pull: |
+      craftctl default
+      case "${CRAFT_TARGET_ARCH}" in
+        i386)
+        ;;
+        *)
+          wget "${SHA256}" "${SIG}" "${URL}"
+          gpg --no-default-keyring \
+              --keyring ./cd-image-keying.gpg \
+              --verify SHA256SUMS.gpg SHA256SUMS
+          awk -v "file=${BASE}" '$2=="*"file' SHA256SUMS | sha256sum -c
+        ;;
+      esac
+    override-build: |
+      rm -rf "${CRAFT_PART_INSTALL}/build-base"
+      mkdir -p "${CRAFT_PART_INSTALL}/build-base"
+      case "${CRAFT_TARGET_ARCH}" in
+        i386)
+          fakeroot debootstrap --arch=i386 --include "$(echo ${DEPENDENCIES} | sed 's/ /,/g')" bullseye "${CRAFT_PART_INSTALL}/build-base"
```

Comment:
Strong NACK.

Ubuntu provides i386 buildd chroots and toolchains to build i386 binaries in all Ubuntu releases, including those that no longer support booting into i386 kernel or running a complete i386 userspace.

Please use Ubuntu to build any i386 binaries. Note it is trivial to do so on amd64 with multiarch repositories enabled. Maybe snapcraft is not enabling i386 for us?

Separately, can you please explain what i386 binaries are being built here?